flojoy/flojoy_cloud.py

```diff
@@ -0,0 +1,409 @@
+        data: Optional[DataC]
+
+        @validator("dataContainer")
+        def type_must_match(cls, dc):
+            assert dc["type"] == "Scalar", "dataContainer type does not match."
+            return dc
+
+        @validator("dataContainer")
+        def keys_must_match(cls, dc):
+            assert "c" in dc, 'dataContainer does not contain "m" dataset.'
+            # int as well just in case?
+            assert isinstance(dc["c"], float), '"c" dataset is not a float'
+            return dc
+
+    class ImageModel(DefaultModel[DataC], Generic[DataC]):
+        data: Optional[DataC]
+
+        @validator("dataContainer")
+        def type_must_match(cls, dc):
+            assert dc["type"] == "Image", "dataContainer type does not match."
+            return dc
+
+        @validator("dataContainer")
+        def keys_must_match(cls, dc):
+            assert "r" in dc, 'dataContainer does not contain "r" dataset.'
+            assert "g" in dc, 'dataContainer does not contain "r" dataset.'
+            assert "b" in dc, 'dataContainer does not contain "r" dataset.'
+            assert isinstance(dc["r"], list), '"r" dataset is not a list'
+            assert isinstance(dc["g"], list), '"g" dataset is not a list'
+            assert isinstance(dc["b"], list), '"b" dataset is not a list'
+            return dc
+
+    dc_type = response["dataContainer"]["type"]
+    match dc_type:
+        case "OrderedPair":
+            try:
+                OrderedPairModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "OrderedTriple":
+            try:
+                OrderedTripleModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "DataFrame":
+            try:
+                DataFrameModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Matrix":
+            try:
+                MatrixModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Grayscale":
+            try:
+                GrayscaleModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Scalar":
+            try:
+                ScalarModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Image":
+            try:
+                ImageModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+
+
+class FlojoyCloud:
+    """
+    A class that allows pulling and pushing DataContainers from the
+    Flojoy cloud client (cloud.flojoy.ai).
+
+    Returns data in a pythonic format (e.g. Pillow for images,
+    DataFrames for arrays/matrices).
+
+    Will support the majority of the Flojoy cloud API:
+    https://rest.flojoy.ai/api-reference
+    """
+
+    def __init__(self, apikey="default", content="application/json"):
```

Comment:
Is the `content` parameter needed? Will the user ever want any data back that isn't json?

Reply:
Not sure but it was in all the REST examples at rest.flojoy.ai

Reply:
@dstrande We can remove that content parameter, also I don't think we should have a default value for apiKey.